infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -396,6 +394,15 @@ private boolean runRecoveryReconfiguration() {
         getCorfuRuntime().invalidateLayout();
         Layout clusterLayout = getCorfuRuntime().getLayoutView().getLayout();
 
+        // Trigger sequencer recovery only if this node was successful in committing the recovery
```

Comment:
I don't understand this change. recoverCluster can trigger sequencer recovery, why do you need to call triggerSequencerBootstrap again?

The issue I opened was related to running the sequencer reconfiguration twice. One through runRecoveryReconfiguration and then bootstrapPrimarySequencerServer, I think.

How does this eliminate the second run ?

Reply:
recoverCluster does not triggerSequencerRecovery. We removed it in the previous change.
We were seeing 2 instances of the sequencer recovery because the first time (after recovery) the sequencerBootstrap API used to fail because of wrongEpochException as we did not save the latest layout to the local datastore. Since the sequencer is still unbootsrtrapped, another task would be triggered later.